newrelic/resource_newrelic_synthetics_monitor.go

```diff
@@ -95,158 +94,411 @@ func resourceNewRelicSyntheticsMonitor() *schema.Resource {
-		monitor.Options.ValidationString = validationString.(string)
+	if v, ok := d.GetOk("locations_public"); ok {
+		simpleMonitorInput.Locations.Public = expandSyntheticsSimplePublicLocations(v.(*schema.Set).List())
 	}
 
-	if verifySSL, ok := d.GetOkExists("verify_ssl"); ok {
-		monitor.Options.VerifySSL = verifySSL.(bool)
+	if v, ok := d.GetOk("locations_private"); ok {
+		simpleMonitorInput.Locations.Private = expandSyntheticsSimplePrivateLocations(v.(*schema.Set).List())
 	}
 
-	if bypassHeadRequest, ok := d.GetOkExists("bypass_head_request"); ok {
-		monitor.Options.BypassHEADRequest = bypassHeadRequest.(bool)
+	if v, ok := d.GetOk("treat_redirect_as_failure"); ok {
+		simpleMonitorInput.AdvancedOptions.RedirectIsFailure = v.(bool)
 	}
 
-	if treatRedirectAsFailure, ok := d.GetOkExists("treat_redirect_as_failure"); ok {
-		monitor.Options.TreatRedirectAsFailure = treatRedirectAsFailure.(bool)
+	if v, ok := d.GetOk("validation_string"); ok {
+		simpleMonitorInput.AdvancedOptions.ResponseValidationText = v.(string)
 	}
 
-	monitor.Locations = locations
-	return &monitor
-}
+	if v, ok := d.GetOk("bypass_head_request"); ok {
+		simpleMonitorInput.AdvancedOptions.ShouldBypassHeadRequest = v.(bool)
+	}
 
-func readSyntheticsMonitorStruct(monitor *synthetics.Monitor, d *schema.ResourceData) {
-	_ = d.Set("name", monitor.Name)
-	_ = d.Set("type", monitor.Type)
-	_ = d.Set("frequency", monitor.Frequency)
-	_ = d.Set("uri", monitor.URI)
-	_ = d.Set("locations", monitor.Locations)
-	_ = d.Set("status", monitor.Status)
-	_ = d.Set("sla_threshold", monitor.SLAThreshold)
-	_ = d.Set("verify_ssl", monitor.Options.VerifySSL)
-	_ = d.Set("validation_string", monitor.Options.ValidationString)
-	_ = d.Set("bypass_head_request", monitor.Options.BypassHEADRequest)
-	_ = d.Set("treat_redirect_as_failure", monitor.Options.TreatRedirectAsFailure)
+	if v, ok := d.GetOk("verify_ssl"); ok {
+		simpleMonitorInput.AdvancedOptions.UseTlsValidation = v.(bool)
+	}
+	return simpleMonitorInput
 }
 
+// TODO: Reduce the cyclomatic complexity of this func
+// nolint:gocyclo
 func resourceNewRelicSyntheticsMonitorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*ProviderConfig).NewClient
-	monitorStruct := buildSyntheticsMonitorStruct(d)
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
 
-	log.Printf("[INFO] Creating New Relic Synthetics monitor %s", monitorStruct.Name)
+	var diags diag.Diagnostics
 
-	monitor, err := client.Synthetics.CreateMonitorWithContext(ctx, monitorStruct)
-	if err != nil {
-		return diag.FromErr(err)
+	monitorType := d.Get("type")
+
+	switch monitorType.(string) {
+	case "SIMPLE":
+		simpleMonitorInput := buildSyntheticsSimpleMonitor(d)
+
+		resp, err := client.Synthetics.SyntheticsCreateSimpleMonitorWithContext(ctx, accountID, simpleMonitorInput)
+
+		if err != nil {
```

Comment:
The error checking code is duplicate between simple and browser. So I think we can move it outside of the switch case. This will probably help with `cyclomatic complexity`